mava/systems/mat/anakin/mat.py

```diff
@@ -0,0 +1,647 @@
+# Copyright 2022 InstaDeep Ltd. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import copy
+import time
+from functools import partial
+from typing import Any, Dict, Tuple
+
+import chex
+import flax
+import hydra
+import jax
+import jax.numpy as jnp
+import optax
+from colorama import Fore, Style
+from flax.core.frozen_dict import FrozenDict
+from jax import tree
+from jumanji.env import Environment
+from omegaconf import DictConfig, OmegaConf
+from rich.pretty import pprint
+
+from mava.evaluator import ActorState, get_eval_fn
+from mava.networks.mat_network import MultiAgentTransformer
+from mava.systems.mat.types import LearnerState, OptStates, Params
+from mava.systems.ppo.types import PPOTransition
+from mava.types import (
+    ActorApply,
+    CriticApply,
+    ExperimentOutput,
+    LearnerFn,
+    TimeStep,
+)
+from mava.utils import make_env as environments
+from mava.utils.checkpointing import Checkpointer
+from mava.utils.jax_utils import (
+    merge_leading_dims,
+    unreplicate_batch_dim,
+    unreplicate_n_dims,
+)
+from mava.utils.logger import LogEvent, MavaLogger
+from mava.utils.network_utils import get_action_head
+from mava.utils.total_timestep_checker import check_total_timesteps
+from mava.utils.training import make_learning_rate
+from mava.wrappers.episode_metrics import get_final_step_metrics
+
+
+def get_learner_fn(
+    env: Environment,
```

Comment:
Can you do a find and replace all of these with MarlEnv
```suggestion
    env: MarlEnv,
```

Reply:
Done